pkg/controller/healthservice/healthservice.go

```diff
@@ -465,32 +465,37 @@ func (r *ReconcileHealthService) updateHealthServiceIngress(h *operatorv1alpha1.
 
 }
 
-func (r *ReconcileHealthService) desiredHealthServiceIngress(h *operatorv1alpha1.HealthService) *extensionsv1.Ingress {
+func (r *ReconcileHealthService) desiredHealthServiceIngress(h *operatorv1alpha1.HealthService) *networkingv1.Ingress {
 	hsName := healthResourceName
 	labels := labelsForHealthService(hsName, h.Name)
 	annotations := annotationsForHealthServiceIngress()
+	pathType := networkingv1.PathType("ImplementationSpecific")
 
 	reqLogger := log.WithValues("HealthService.Namespace", h.Namespace, "HealthService.Name", h.Name)
 	reqLogger.Info("Building HealthService Ingress", "Ingress.Namespace", h.Namespace, "Ingress.Name", hsName)
 
-	ing := &extensionsv1.Ingress{
+	ing := &networkingv1.Ingress{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        hsName,
 			Namespace:   h.Namespace,
 			Labels:      labels,
 			Annotations: annotations,
 		},
-		Spec: extensionsv1.IngressSpec{
-			Rules: []extensionsv1.IngressRule{
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
 				{
-					IngressRuleValue: extensionsv1.IngressRuleValue{
-						HTTP: &extensionsv1.HTTPIngressRuleValue{
-							Paths: []extensionsv1.HTTPIngressPath{
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
 								{
-									Path: "/cluster-health/",
-									Backend: extensionsv1.IngressBackend{
-										ServiceName: hsName,
-										ServicePort: intstr.IntOrString{Type: intstr.Int, IntVal: 6967}},
+									Path:     "/cluster-health/",
```

Comment:
see my comments about fixed strings, for `/must-gather`

Reply:
Introduced constant file to address these values.